nmstatectl/nmstatectl.py

```diff
@@ -90,12 +116,49 @@ def _filter_interfaces(state, patterns):
     return showinterfaces
 
 
-def show(args):
-    state = netinfo.show()
-    if args.only != '*':
-        patterns = [p for p in args.only.split(',')]
+def _filter_state(state, whitelist):
+    if whitelist != '*':
+        patterns = [p for p in whitelist.split(',')]
         state['interfaces'] = _filter_interfaces(state, patterns)
+    return state
+
+
+def edit(args):
+    state = _filter_state(netinfo.show(), args.only)
```

Comment:
Please check whether specified interface exists:
```python
    if not state['interfaces']:
        sys.stderr.write('ERROR: No such interface\n')
        return 1
```

Reply:
An empty state is not an error.

Reply:
How about making this a warning at least? I agree with @cathay4t that it makes sense to report this so that nmstate does not silently nothing when the user mistyped the interface name.

Reply:
added the error